Backend/routes/Auth.js

```diff
@@ -421,37 +425,76 @@ async function matchedBuddy(date, time, destinationId, meetingPointId, userId) {
       }
     }
 
+    const userPreferences = await prisma.user.findUnique({
+      where: { id: userId },
+      select: { preferences: true, major: true, classification: true },
+    });
+    const totalPreferences =
+      Number(userPreferences.preferences.distance) +
+      Number(userPreferences.preferences.major) +
+      Number(userPreferences.preferences.classification);
+    const distanceWeight =
+      Number(userPreferences.preferences.distance) / totalPreferences;
+    const majorWeight =
+      Number(userPreferences.preferences.major) / totalPreferences;
+    const classificationWeight =
+      Number(userPreferences.preferences.classification) / totalPreferences;
+
     const locationCoordinates = await locationCod(meetingPointId);
     const userLat = locationCoordinates.latitude;
     const userLon = locationCoordinates.longitude;
 
-    const distances = [];
+    const scores = [];
     for (let i = 0; i < filteredBuddies.length; i++) {
-      const locationCoordinatesBuddy = await locationCod(
-        filteredBuddies[i].locationId,
-      );
+      const buddy = filteredBuddies[i];
+      const locationCoordinatesBuddy = await locationCod(buddy.locationId);
       const buddyLat = locationCoordinatesBuddy.latitude;
       const buddyLon = locationCoordinatesBuddy.longitude;
 
       const distance = haversineFormula(userLat, userLon, buddyLat, buddyLon);
 
-      distances.push({ distance, buddy: filteredBuddies[i] });
+      let distanceScore = 0;
```

Comment:
Put this out of loop, preferably at top of your file for constant definition. Otherwise you refine these values every loop